File: src/Piece.cpp
#include "../include/Piece.h"


    Color Piece::getColor() {
        return color;
    } //Color of the piece

    PieceType Piece::getType() {
        return type;
    }  //Type of the piece

    int Piece::getRow() {
        return row;
    } 

    int Piece::getCol() {
        return col;
    } 

    void Piece::newCords(int newRow, int newCol) {
        row = newRow;
        col = newCol;
    }

    void Piece::setRow(int newRow) {
        row = newRow;
    } 

    void Piece::setCol(int newCol) {
        col = newCol;
    } 

    void Piece::setColor(Color newColor){
        color = newColor;
    }

    void Piece::setType(PieceType newType){
        type = newType;
    }

    void Piece::operator=(Piece& other){
        type = other.type;
        color = other.color;
        col = other.col;
        row = other.row;
    }

    void Piece::changePiece(Color newColor, PieceType newType, int newRow, int newCol){
        type = newType;
        color = newColor;
        col = newCol;
        row = newRow;
    }

File: src/Piece.java
/**
 * Класс, представляющий шахматную фигуру
 */
public class Piece {
    private final Core.PieceType type;
    private final Core.Color color;
    private final Position position;

    public Piece(Core.PieceType type, Core.Color color, Position position) {
        if (type == null) {
            throw new IllegalArgumentException("type cannot be null");
        } else if (position == null) {
            throw new IllegalArgumentException("position cannot be null");
        } else if (color == null) {
            throw new IllegalArgumentException("color cannot be null");
        }

        this.type = type;
        this.color = color;
        this.position = position;
    }

    public Core.PieceType getType() {
        return type;
    }

    public Core.Color getColor() {
        return color;
    }

    public Position getPosition() {
        return position;
    }

    @Override
    public String toString() {
        return "Piece{" + "type = " + type + ", color = " + color + ", position = " + position + '}';
    }
}